pytket/pytket/qasm/qasm.py

```diff
@@ -1142,12 +1142,17 @@ def circuit_to_qasm_io(
                     variable = control_bit
                 else:
                     variable = control_bit.reg_name
+                    if hqs_header(header) and bits != list(cregs[variable]):
+                        raise QASMUnsupportedError(
+                            "HQS QASM conditions must be an entire classical register "
```

Comment:
Most pedantic comment over, but maybe `hqs_header QASM conditions...` is more precise these days

Reply:
Is this constraint specified in the hqs_header though? Or is it a hardware restriction that is bounced back via the Quantinuum API or something?

Reply:
I'm actually not sure (I've asked). If it is allowed then we need to do something more clever here -- but meanwhile, better to throw an error than do the _wrong_ thing.

Reply:
> Most pedantic comment over, but maybe `hqs_header QASM conditions...` is more precise these days

Fixed.